### ISSUE TITLE
Sometimes the keys d, f and b do not respond as expected. 

### ISSUE THREAD

**Issue by doewie** [jeelabs/folie#44]
Running Folie v2.10-7-g05bad42 on W10.
I'm not sure this is a W10 related problem but sometimes the d and f keys do not respond and the b key repositions the cursor back to the beginning of the previous word. Only after restart of Folie it's OK again.
Unfortunately I was unable to figure out when this is happening. The problem usually does not occur.

**Comment by jcw** [comment]
That's an odd one - sounds as if the software thinks that the CTRL key is pressed, in which case it's definitely a bug in https://github.com/chzyer/readline ...

**Comment by rolfhand** [comment]
I'm also using W10 now. Latest Folie release and FTDI connections to 2 BluePills with RFM69HW.
After 1 to 10 minutes folie stops accepting some lowercase characters ( d and f at least. Capital characters keep working all the time so there is no real problem. Cursor movement is strange then and <delete> does not work, backspace does.  These issues did not occur when working in an ssh session on a raspberry running folie.

**Comment by jcw** [comment]
Thanks for the info. We really need to get to the bottom of this.

Could either one of you try the "mini folie" I mentioned #46? - it uses readline in more mainstream mode, and has none of the extra features (in fact, it's all in a single main.go source file). It may be easier to diagnose and chase this bug when we move all the extras out of the way.

**Comment by rolfhand** [comment]
I'll find some time next weekend. Do you use a 1.7 version, then i wiil upgrade first.

**Comment by jcw** [comment]
Thanks. Yes, I'm on 1.7.5 - but I _assume_ that older versions of Go will work as well.

**Comment by doewie** [comment]
FYI: i'm testing on go1.7.5 windows/amd64

**Comment by rolfhand** [comment]
@doewie  Another approach. I set  my commandwindow in UTF-8  "chcp 65001".It seems  to work. Can you try that too?

**Comment by rolfhand** [comment]
Tried  the minifolie today  (go run main.go), same errors. Lower case **d** was the first character to disappear. Characters b, d, f and g ( if i'm correct)are acting like some cursor commands.

**Comment by ctmalan** [comment]
Ditto.  I have the same bug in Win7.  Alt-tab seems to cause it sometimes, not sure, it happens when switching to Sublime.

Lowercase commands and ! goes missing as reported.  Backspace also becomes unreliable.

**Comment by jcw** [comment]
Please retry with the new Folie v2.11. This is a really weird issue. We need to nail and fix it.

**Comment by doewie** [comment]
OK, just installed v2.11. Will do some tests coming days.

**Comment by ctmalan** [comment]
Is a win86 binary available?

c-:

**Comment by jcw** [comment]
I think so: "folie-windows.gz" on the [Releases](https://github.com/jeelabs/folie/releases) page?

**Comment by ctmalan** [comment]
Cannot find exe inside?  Sorry...

**Comment by jcw** [comment]
Whoops! My fault, the files are exe's - please either rename the one you have, or download again - I've now fixed the 32- and 64-bit Windows ZIP files.

**Comment by ctmalan** [comment]
Dear Jean-Claude,

My antivirus quarantined the file? Any idea why?

c-:

**Comment by ctmalan** [comment]
Found Gen:Variant.Zusy.233235

**Comment by ctmalan** [comment]
Possibly a Bitdefender false alarm... reported at Bitdefender.

**Comment by ctmalan** [comment]
New funny:  (are you sure this is not a virus?)

!send xman.fs
open xmem.fs: The system cannot find the file specified.
Send failed.
!send xman.fs
open xmem.fs: The system cannot find the file specified.
Send failed.

**Comment by jcw** [comment]
The virus alarm may be due to the fact that Go tends to generate 100% self-contained exe's. I've seen it happen before.

That send failing is as puzzling as the asdf keys not being recognised. Something totally weird seems to be going on here!

Maybe @doewie can shed some light on this...

**Comment by jcw** [comment]
it's tagged as a virus not just by Bitdefender - https://www.virustotal.com/en/file/5872144f1adbf59597b10c1e7b56a3a969017026cb2cb3c1fe123b159f56b971/analysis/1494061227/

**Comment by ctmalan** [comment]
Please delete "New funny:" entry, this was operator brain failure!

c-:

**Comment by jcw** [comment]
You should be able to edit this yourself - click the pencil or cross on [this page](https://github.com/jeelabs/folie/issues/44).

**Comment by doewie** [comment]
Testing on folie 2.11 the problem still exists. 
I'm almost sure it happens after ALT-TAB to another window, and then ALT-TAB back to the Folie screen.
Maybe another Windows-user can confirm this.

**Comment by ctmalan** [comment]
Also happens after alt-tab here, but not every time.

**Comment by ctmalan** [comment]
Bitdefender cleared the false alarm - now testing with 2.11 ...

**Comment by ctmalan** [comment]
Now fails reliably on my system.  Do as follows: Run from shortcut:

<path>\folie.exe -r -p COM38 -b 115200

test by typing...
sdfb sdfb not found.

alt-tab to Chrome
alt-tab back
s s not found

Lower case dfb goes missing with:
f-> move to end of line
b-> move to start of line
d-> delete from here

Hope this helps to solve the issue...

**Comment by jcw** [comment]
Thanks for chasing this down, great progress.

But I'm _totally_ in the dark about what's going on :( - how on earth can some _keys_ get gobbled. We may have to slice up the app and remove more and more to chase this one.

**Comment by jcw** [comment]
Seems to me that Folie is thinking a modifier key is currently active. Could you try something?

After switching focus back, press each of shift, ctrl, alt and release them again, _then_ try "s" etc.

**Comment by doewie** [comment]
You're right! After pressing ALT the f,d and b-keys are back again. So indeed it looks as if Folie thinks the ALT is still active after tabbing out and in again.

**Comment by ctmalan** [comment]
We could document this as a edit modifier in the help file.  Thus no change
to the code is required.

:-?
